compiler/rustc_trait_selection/src/traits/query/type_op/mod.rs

```diff
@@ -81,16 +81,14 @@ pub trait QueryTypeOp<'tcx>: fmt::Debug + Sized + TypeFoldable<'tcx> + 'tcx {
         // like the subtype query, which go awry around
         // `'static` otherwise.
         let mut canonical_var_values = OriginalQueryValues::default();
-        let canonical_self =
-            infcx.canonicalize_hr_query_hack(&query_key, &mut canonical_var_values);
+        let old_param_env = query_key.param_env;
+        let canonical_self = infcx.canonicalize_hr_query_hack(query_key, &mut canonical_var_values);
         let canonical_result = Self::perform_query(infcx.tcx, canonical_self)?;
 
-        let param_env = query_key.param_env;
-
         let InferOk { value, obligations } = infcx
             .instantiate_nll_query_response_and_region_obligations(
                 &ObligationCause::dummy(),
-                param_env,
+                old_param_env,
```

Comment:
do we intentionally use the old param_env here instead of `canonical_self.value.param_env`?

Reply:
I believe so. This is I think the parameter environment that we are "instantiating the response into", so to speak.